### PR TITLE
Add COMMENT ON statement

### PR DESCRIPTION
**Pull request by samansmink**
This PR adds a new feature to DuckDB: Comments.

Various [other](https://docs.snowflake.com/en/sql-reference/sql/comment) [databases](https://docs.aws.amazon.com/redshift/latest/dg/r_COMMENT.html) also support this, we follow the [Postgres](https://www.postgresql.org/docs/current/sql-comment.html) syntax in this PR:

Comments are currently only string literals or NULL, and can be set using:
```
COMMENT ON <type> <name> IS 'string literal'
COMMENT ON <type> <name> IS NULL
```

## What can I comment on?
```SQL
COMMENT ON TABLE test_table IS 'very gezellige table';
COMMENT ON COLUMN test_table.test_table_column IS 'very gezellige column';
COMMENT ON VIEW test_view IS 'very gezellige view';
COMMENT ON INDEX test_index IS 'very gezellige index';
COMMENT ON SEQUENCE test_sequence IS 'very gezellige sequence';
COMMENT ON TYPE test_type IS 'very gezellige type';
COMMENT ON MACRO test_macro IS 'very gezellige macro';
COMMENT ON MACRO TABLE test_table_macro IS 'very gezellige table macro' ;
```

## How to view comments?

```SQL
SELECT comment FROM duckdb_tables();
SELECT comment FROM duckdb_columns();
SELECT comment FROM duckdb_views();
SELECT comment FROM duckdb_indexes();
SELECT comment FROM duckdb_sequences();
SELECT comment FROM duckdb_types();
SELECT comment FROM duckdb_functions();
SELECT comment FROM duckdb_functions();
```
## Limitations
- Currently you can not comment on schemas or databases.
- Also you can't comment on things that have a dependency (e.g. a table with an index)

Small note to @tishj this also fixes a very minor pyspark bug where the incorrect field of the duckdb_columns() was read and would return the incorrect value for nullable

### CODE REVIEW

**Comment by samansmink** [comment]
Does not pass CI yet due to breaking with some extensions, but I would like to get a review before fixing that!

**Comment by Mause** [comment]
It would be great if we could populate the `pg_catalog.pg_description.description` field with the comment at the same time as well: https://github.com/duckdb/duckdb/blob/5b391512a8fc7800a0c22d98714e6d8f25c002cf/src/catalog/default/default_views.cpp#L36

File: src/parser/expression/columnref_expression.cpp [hunk 1]
```diff
@@ -51,6 +51,19 @@ const string &ColumnRefExpression::GetTableName() const {
 	return column_names[0];
 }
 
+const string &ColumnRefExpression::GetSchemaName() const {
+	D_ASSERT(column_names.size() >= 3 && column_names.size() <= 4);
+	if (column_names.size() == 4) {
+		return column_names[1];
```
Comment (Mytherin) [inline src/parser/expression/columnref_expression.cpp:57]: We have a very flexible dot syntax - we support all of `schema.table.column`, `catalog.table.column`, `table.column.struct_entry`, `column.struct_entry.nested_struct_entry`. As such `GetSchemaName()` and `GetCatalogName()` as they are implemented here are rather confusing and can be wrong.

Could you remove these from the `ColumnRefExpression`?

File: test/sql/catalog/comment_on_pg_description.test [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+statement ok
+COMMENT ON COLUMN test.a IS 'comment-2'
+
+statement ok
+CREATE VIEW test_view AS FROM test;
+
+# TODO comment on view column
+#statement ok
+#COMMENT ON COLUMN test_view.a IS 'comment-2.5'
```
Comment (Mytherin) [inline test/sql/catalog/comment_on_pg_description.test:19]: Can we do this with `mode skip/mode unskip`?

File: test/sql/catalog/comment_on_extended.test [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+0
+
+query I
+SELECT count(*) from duckdb_columns() where comment IS NOT NULL;
+----
+0
+
+### Tables with fully qualified names
+statement ok
+COMMENT ON TABLE db1.s1.t1 IS 'very gezellige table 1'
```
Comment (Mytherin) [inline test/sql/catalog/comment_on_extended.test:52]: Can we also test partially qualified names that are unambiguous, e.g. `COMMENT ON TABLE db1.t1` and `COMMENT ON TABLE s1.t1`?

File: src/main/database_manager.cpp [hunk 4]
```diff
@@ -73,6 +73,22 @@ void DatabaseManager::DetachDatabase(ClientContext &context, const string &name,
 	}
 }
 
+optional_ptr<AttachedDatabase> DatabaseManager::AlterDatabase(ClientContext &context, AlterInfo &info) {
```
Comment (Mytherin) [inline src/main/database_manager.cpp:76]: Is this actually used or is this a left-over from a previous impl?

**Review by Mytherin — COMMENTED** [review]
Thanks for the changes! Looks great - some more minor comments:

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**